apps/api/src/app/partner-integrations/usecases/update-vercel-configuration/update-vercel-configuration.usecase.ts

```diff
@@ -0,0 +1,261 @@
+export class UpdateVercelConfiguration {
+  constructor(
+    private httpService: HttpService,
+    private environmentRepository: EnvironmentRepository,
+    private getVercelProjectsUsecase: GetVercelProjects,
+    private organizationRepository: OrganizationRepository
+  ) {}
+
+  async execute(command: UpdateVercelConfigurationCommand): Promise<{ success: boolean }> {
+    try {
+      const organizationIds = Object.keys(command.data);
+      const projectIds = Object.keys(command.data).reduce((acc, curr) => {
+        return acc.concat(command.data[curr]);
+      }, []);
+
+      const envKeys = await this.getEnvKeys(organizationIds);
+
+      const configurationDetails = await this.getVercelProjectsUsecase.getVercelConfiguration({
+        configurationId: command.configurationId,
+        userId: command.userId,
+      });
+
+      const { addProjectIds, updateProjectDetails } = await this.getVercelProjects(
+        configurationDetails.accessToken,
+        configurationDetails.teamId,
+        projectIds
+      );
+
+      const mappedProjectData = this.mapProjectKeys(envKeys, command.data, { updateProjectDetails, addProjectIds });
+
+      for (const key of Object.keys(mappedProjectData)) {
+        await this.setEnvironments({
+          clientKey: mappedProjectData[key].clientKey,
+          privateKey: mappedProjectData[key].privateKey,
+          projectIds: mappedProjectData[key].addProjectIds,
+          teamId: configurationDetails.teamId,
+          token: configurationDetails.accessToken,
+        });
+        await this.updateEnvironments({
+          clientKey: mappedProjectData[key].clientKey,
+          privateKey: mappedProjectData[key].privateKey,
+          projectDetails: mappedProjectData[key].updateProjectDetails,
+          teamId: configurationDetails.teamId,
+          token: configurationDetails.accessToken,
+        });
+      }
+
+      await this.saveProjectIds(command);
+
+      return {
+        success: true,
+      };
+    } catch (error) {
+      throw new ApiException(error.message);
+    }
+  }
+
+  private async getEnvKeys(organizationIds: string[]): Promise<EnvironmentEntity[]> {
+    return await this.environmentRepository.find(
+      {
+        _organizationId: organizationIds,
+      },
+      'apiKeys identifier _organizationId'
+    );
+  }
+
+  private mapProjectKeys(
+    envData: EnvironmentEntity[],
+    projectData: Record<string, string[]>,
+    projectDetails: MappedProjectData
+  ) {
+    const { addProjectIds, updateProjectDetails } = projectDetails;
+    const mappedData = envData.reduce<Record<string, MapProjectkeys>>((acc, curr) => {
+      const projectIds = projectData[curr._organizationId];
+      const newData = {
+        privateKey: curr.apiKeys[0].key,
+        clientKey: curr.identifier,
+        updateProjectDetails: updateProjectDetails.filter((detail) => projectIds.includes(detail.projectId)),
+        addProjectIds: projectIds.filter((id) => addProjectIds.includes(id)),
```

Comment:
`updateProjectDetails` will be the projects that were previously added during integration setup, and now updated during the update configuration.
e.g, someone switched `Project A`  <-> `org 1` to `Project A` <-> `org 2`

`addProjectIds` are the new projects that the user added during the update configuration phase